tests/javascript/Tiltfile

```diff
@@ -0,0 +1,59 @@
+# Some common test configurations to easily get up and running with Tilt for Tests
+# (see docs: https://docs.tilt.dev/tests_in_tilt.html)
+
+
+def _test_js(name, pkmanager, dir, deps=None, with_install=False, project_root='', ignore=None, extra_args=None):
+    if not project_root:
+        project_root = dir
+
+    install_deps = deps_for_package_manager(pkmanager, project_root)
+    resource_deps = []
+    if with_install:
+        install_resource = pkmanager + '-install'
+        install_cmd = 'cd {dir} && {pkmanager} install'.format(
+            dir=dir, pkmanager=pkmanager)
+        local_resource(install_resource, install_cmd, install_deps, allow_parallel=True)
+        resource_deps = [install_resource]
+
+        # double check that we could find package.json etc. correctly
+        for f in install_deps:
+            if not os.path.exists(f):
+                warn('''expected file "{}" not found -- did you pass project_root correctly when setting up JS tests?
+    If not, we can't watch this file for changes and reinstall JS deps when necessary.'''.format(f))
```

Comment:
> though really I'd probably change the second half to "automatic node module updating disabled" or so. As written it kinda feels like "you screwed up and you need to fix it or else this".
do you think i should DISABLE the updating if i can't find the pkg.json/lockfile? currently all this means is that we can't watch those files to trigger a re-install--but we should be able to run the `yarn|npm install` from anywhere in the project

Reply:
> currently all this means is that we can't watch those files to trigger a re-install

> we should be able to run the yarn|npm install from anywhere in the project

I'm not sure if I'm misunderstanding the behavior when these files are missing or if I'm misunderstanding the above comment.

My understanding is that if the user specifies the wrong dir, Tilt will not be watching the right paths, and so will not automatically re-run yarn/npm install.

What does "should be able to run the yarn|npm install from anywhere in the project" mean? I'm not sure of an interpretation other than "manually trigger", which wouldn't be an automatic update.

Or are there other deps I'm missing that would trigger automatic updates?

Reply:
> What does "should be able to run the yarn|npm install from anywhere in the project" mean? I'm not sure of an interpretation other than "manually trigger", which wouldn't be an automatic update.

i just mean that we can still do the _installing_, if not the watching+auto-rerun, if the user doesn't give us the correct project_root. e.g. in Tilt if i specify `dir=./web/src`, i won't be able to find `package.json` (it lives in `./web`) but i WILL be able to run `yarn install` from there

so yeah the question right now is, if the user doesn't pass `project_root` and `$DIR/package.json` etc. doesn't exist, do we:
1. run the install cmd (which we can still do) and warn the user that we can't watch `package.json` etc. and so Tilt won't automatically re-run yarn/npm install if deps files change, OR
2. assume that something is pretty wrong and error out
?

thinking about this now, option 2 is maybe safer -- i guess a user can always write their own dependency-installer local resource if their setup is too weird?

Reply:
Ah ok I meant "automatic updating" as in "automatically run on file changes", but it was ambiguous in that it could also be read as "automatically trigger on startup". Makes sense!

but yeah I feel like option 2 is simplest

Reply:
cool, done